src/actions/actions/services/dispatchers/verusid/verusid.js

```diff
@@ -56,3 +65,193 @@ export const unlinkVerusId = async (iAddress, chain) => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning');
+  }
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+  const currentPendingIdentities =
+    serviceData.pending_ids == null ? {} : serviceData.pending_ids;
+
+  if (currentPendingIdentities[chain]) {
+    delete currentPendingIdentities[chain][iAddress];
+  }
+
+  return await modifyServiceStoredDataForUser(
+    {
+      ...serviceData,
+      pending_ids: currentPendingIdentities,
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const deleteAllProvisionedIds = async () => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning');
+  }
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+
+  return await modifyServiceStoredDataForUser(
+    {
+      ...serviceData,
+      pending_ids: {},
+    },
+    VERUSID_SERVICE_ID,
+    state.authentication.activeAccount.accountHash,
+  );
+};
+
+export const checkVerusIdNotificationsForUpdates = async () => {
+  const state = store.getState();
+
+  if (state.authentication.activeAccount == null) {
+    throw new Error('You must be signed in for ID provisioning functions');
+  }
+  const isTestnet = Object.keys(state.authentication.activeAccount.testnetOverrides).length > 0;
+
+  const system = isTestnet ? coinsList.VRSCTEST : coinsList.VRSC;
+  const ticker = system.id;
+
+  // Itterate through all pending IDs and check for updates
+  const pendingIds = state.channelStore_verusid.pendingIds;
+
+  const serviceData = await requestServiceStoredData(VERUSID_SERVICE_ID);
+  const currentLinkedIdentities = Object.keys(serviceData?.linked_ids[ticker] || []);
+
+
+  if (pendingIds[ticker] !== null) {
+
+    const details = Object.keys(pendingIds[ticker]);
+    for (const iaddress of details) {
+
+      // once an ID is linked, remove it from pending IDs, or if the server has rejected it delete.
+      if((pendingIds[ticker][iaddress].status === NOTIFICATION_TYPE_VERUSID_READY && 
+         currentLinkedIdentities.indexOf(iaddress) > -1) ||(pendingIds[ticker][iaddress].status === NOTIFICATION_TYPE_VERUSID_FAILED)  ) {
+          await deleteProvisionedIds(iaddress, ticker);
+          await updatePendingVerusIds();
+          continue;
+      } 
+
+      if ((pendingIds[ticker][iaddress].createdAt + 600) < Math.floor(Date.now() / 1000)) {
+        // If the request is older than 10 minutes, check info endpoint to see if it was accepted or rejected
+        let errorFound = false;
+        try {
+          if (pendingIds[ticker][iaddress].infoUri) {
+            const response = await axios.post(pendingIds[ticker][iaddress].infoUri);
+
+            if (response.data.result.state === primitives.LOGIN_CONSENT_PROVISIONING_RESULT_STATE_FAILED.vdxfid) {
```

Comment:
response.data should not be looked at here without it being wrapped in a `const responseData = new LoginConsentProvisioningResponse(response);` as to ensure that the response from the server is formatted correctly